### PR TITLE
Implemented putnbtctx function and test

### PR DESCRIPTION
**Pull request by Rcc999**
## Description

Closes: #184 

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
    
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] added appropriate labels to the PR
- [ ] provided a link to the relevant issue or specification
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included doc comments for public functions
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary

## Summary by Sourcery

Implement putNbtcTx method to detect and register nBTC minting transactions with duplicate checks and network validation, and add corresponding unit tests.

New Features:
- Add Indexer.putNbtcTx to parse transaction hex, identify nBTC deposits, and register new minting records in the database

Enhancements:
- Prevent duplicate processing by checking existing records before insertion
- Validate the specified Bitcoin network and throw on unsupported networks

Tests:
- Add tests to verify successful registration of new nBTC transactions, duplicate handling, and error thrown for transactions without valid deposits

### CODE REVIEW

File: packages/btcindexer/src/btcindexer.test.ts [hunk 1]
```diff
@@ -377,6 +377,56 @@ describe("Block Parsing", () => {
 	});
 });
 
+describe("Indexer.putNbtcTx", () => {
+	it("should return true and register a new tx with a single deposit in db", async () => {
+		const blockData = REGTEST_DATA[329]!;
+		const block = Block.fromHex(blockData.rawBlockHex);
+		const targetTx = block.transactions?.find((tx) => tx.getId() === blockData.txs[1]!.id);
+		expect(targetTx).toBeDefined();
```
[the remark this answers is not in the document]
Comment (Rcc999) [reply packages/btcindexer/src/btcindexer.test.ts]: done

**Review by robert-zaremba — COMMENTED** [review]
- update the PR by removing  `putNbtcTx` and improve `registerBroadcastedNbtcTx`
- check if we already have similar tests for registerBroadcastedNbtcTx. If not , then update the tests accordingly.

File: packages/btcindexer/src/btcindexer.ts [hunk 2]
```diff
@@ -94,15 +94,19 @@ export class Indexer {
...
+		const existingTx = await this.storage.getNbtcMintTx(txId);
+		if (existingTx) {
+			logger.debug({
+				msg: "Tx already exists in db",
+				method: "Indexer.putNbtcTx",
+				txId,
+			});
+			return false;
+		}
+		await this.registerBroadcastedNbtcTx(txHex, network);
```
[the remark this answers is not in the document]
Comment (Rcc999) [reply packages/btcindexer/src/btcindexer.ts]: Alright I see, it is done now
